.github/workflows/cache.yml

```diff
@@ -0,0 +1,58 @@
+# This workflow caches crate dependencies and build artifacts for tools (except 'test-tools' since we don't use them in build workflows).
+# The cache is only usable by workflows started from pull requests against the develop branch.
+name: CacheDepsAndTools
+on:
+  push:
+    branches: [develop]
+    paths:
+      - '.github/**'
+      - 'sources/Cargo.lock'
+      - 'tools/buildsys/**'
+      - 'tools/pubsys*/**'
+      - '!tools/pubsys/policies/**'
+      - '!tools/pubsys/**.example'
+      - '!tools/pubsys/**.template'
+      - 'tools/Cargo.lock'
+jobs:
+  cache:
+    runs-on:
+      group: bottlerocket
+      labels: bottlerocket_ubuntu-latest_16-core
+    continue-on-error: true
+    steps:
+      - run: |
+          echo "OS_ARCH=`uname -m`" >> $GITHUB_ENV
+      - uses: actions/checkout@v3
+      # Cache `cargo-make`, `cargo-cache`, `cargo-sweep`
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo
+          key: ${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}
```

Comment:
`.github/workflows/cache.yaml` shouldn't be used as the exclusive hash file. Changes to this file will cause caches to be rebuilt that don't have any association to this file.

Doesn't it make more sense to use some `Cargo.lock` as a hash file since that is what will change as dependencies change?

I'd imagine this will frequently result in a incorrect hits since `.github/workflows/cache.yaml` won't change.

Reply:
> Cache `cargo-make`, `cargo-cache`, `cargo-sweep`

Or is the idea that this _should_ result in a frequent hit since `cargo-make`, `cargo-cache`, and `cargo-sweep` are tools that are directly associated with this workflow .. ? I'm not understanding something here 🤔